Cannot start opencode with omos v0.9.13

- my opencode.jsonc
```
"plugin": [
		"oh-my-opencode-slim@latest",
]
```

- exec `rm -rf ~/.cache/opencode` to force update plugins, and opencode works fine.
after a while, when i use opencode in a new terminal tab, error occurs with a log: `ERROR 2026-04-16T08:19:19 +172ms service=plugin path=oh-my-opencode-slim@latest target=file:///Users/xxxx/.cache/opencode/packages/oh-my-opencode-slim@latest/node_modules/oh-my-opencode-slim/dist/index.js error=Cannot find module '/Users/alvin/repos/oh-my-opencode-slim/node_modules/jsdom/lib/jsdom/living/xhr/xhr-sync-worker.js' from '/Users/xxxx/.cache/opencode/packages/oh-my-opencode-slim@latest/node_modules/oh-my-opencode-slim/dist/index.js' failed to load plugin`
- in ~/.cache/opencode/packages/oh-my-opencode-slim@latest/package.json
```
{
  "dependencies": {
    "oh-my-opencode-slim": "0.9.13"
  }
}
```

--- a/package.json
+++ b/package.json
@@ -36,8 +36,8 @@
     "LICENSE"
   ],
   "scripts": {
-    "build:plugin": "bun build src/index.ts --outdir dist --target node --format esm --external @ast-grep/napi --external @opencode-ai/plugin --external @opencode-ai/sdk",
-    "build:cli": "bun build src/cli/index.ts --outdir dist/cli --target node --format esm --external @ast-grep/napi --external @opencode-ai/plugin --external @opencode-ai/sdk",
+    "build:plugin": "bun build src/index.ts --outdir dist --target node --format esm --external @ast-grep/napi --external @opencode-ai/plugin --external @opencode-ai/sdk --external jsdom",
+    "build:cli": "bun build src/cli/index.ts --outdir dist/cli --target node --format esm --external @ast-grep/napi --external @opencode-ai/plugin --external @opencode-ai/sdk --external jsdom",
     "build": "bun run build:plugin && bun run build:cli && tsc --emitDeclarationOnly && bun run generate-schema",
     "prepare": "bun run build",
     "contributors:add": "all-contributors add",

--- a/src/tools/smartfetch/tool.ts
+++ b/src/tools/smartfetch/tool.ts
@@ -448,7 +448,11 @@ export function createWebfetchTool(
                     decodeBody(body.data, headerMetadata.charset, contentType);
                   const rawText = decoded.text;
                   const extracted = isHtmlLikeContentType(contentType)
-                    ? extractFromHtml(rawText, finalUrl, args.extract_main)
+                    ? await extractFromHtml(
+                        rawText,
+                        finalUrl,
+                        args.extract_main,
+                      )
                     : {
                         title: undefined,
                         rawContent: cleanFetchedText(rawText),

--- a/src/tools/smartfetch/utils.ts
+++ b/src/tools/smartfetch/utils.ts
@@ -1,7 +1,14 @@
 import { Readability } from '@mozilla/readability';
-import { JSDOM } from 'jsdom';
 import TurndownService from 'turndown';
 import type { CachedFetch, ExtractedContent } from './types';
+
+let jsdomPromise: Promise<typeof import('jsdom')> | undefined;
+
+async function getJSDOM() {
+  jsdomPromise ??= import('jsdom');
+  const { JSDOM } = await jsdomPromise;
+  return JSDOM;
+}
 
 export function wordCount(text: string) {
   const trimmed = text.trim();
@@ -276,11 +283,12 @@ turndown.addRule('fenced-pre-code', {
   },
 });
 
-export function extractFromHtml(
+export async function extractFromHtml(
   html: string,
   finalUrl: string,
   extractMain: boolean,
-): ExtractedContent {
+): Promise<ExtractedContent> {
+  const JSDOM = await getJSDOM();
   const dom = new JSDOM(html, { url: finalUrl });
   const document = dom.window.document;
   const title = document.title || undefined;

